cmd/ctr/commands/tasks/start.go

```diff
@@ -85,7 +87,12 @@ var startCommand = cli.Command{
 			}
 		}
 
-		task, err := NewTask(ctx, client, container, "", con, context.Bool("null-io"), ioOpts, opts...)
+		task, _ := container.Task(ctx, nil)
```

Comment:
I think i'd prefer the logic to be:

```go
if _, err := container.Task(ctx, nil); err == nil {
    return fmt.Errorf("task %s already exists", id)
}
```

Reply:
Shouldn't `NewTask` just return `err` `"task with id: %s already exists"`? This just shortens a race but doesn't really avoid it.

Reply:
This is also true and the correct way to handle this.

Reply:
```
if _, err := container.Task(ctx, nil); err == nil {
    return fmt.Errorf("task %s already exists", id)
}
```
This is more simple but I think error message change to `fmt.Errorf("cannot start a already exist task %s", id)` can be more exact.